tests/core/go_test/testmain_bad_test.go

```diff
@@ -0,0 +1,13 @@
+// +build !good
+
+package testmain_bad
+
+import "testing"
+
+func TestMain(m *testing.M) {
```

Comment:
It looks like the correct behavior is for this test to always fail. That will break CI, since we essentially run `bazel test //...`.

Please convert this to a `go_bazel_test` that sets up a simple workspace, runs `bazel test` on a target, and verifies that the test fails. See `test_filter_test` for a simple example. Specifically, check that Bazel's exit code is `bazel_testing.TESTS_FAILED`.

Reply:
Yeah, I knew it was supposed to fail, didn't know how to test it (thought the `// +build !good` is the trick, see comment above).

Thanks for the pointer.